test/integration/database-helper_test.js

```diff
@@ -114,29 +113,31 @@ describe('Integration | db-connection.js', () => {
       expect(skillIdsCount).to.deep.equal(skillCount);
     });
 
-    it('should throw an error if the data to be inserted has no primary key', async function() {
-      const table = {
-        name: 'challenges',
-        fields: [
-          { name: 'instructions', type: 'text' },
-          { name: 'timer', type: 'smallint' },
-          { name: 'autoReply', type: 'boolean' },
-          { name: 'skillIds', type: 'text []', isArray: true },
-          { name: 'skillCount', type: 'smallint', extractor: (record) => _.size(record['skillIds']) },
-          { name: 'firstSkillId', type: 'text', extractor: (record) => _.get(record['skillIds'], 0) },
-        ],
-        indices: ['firstSkillId'],
-      };
-      const fullLearningContent = {
-        challenges: [{}],
-      };
-      const learningContent = fullLearningContent[table.name];
+    context('when at least one of the collection item has no property', () => {
+      it('should throw an error', async function() {
+        const table = {
+          name: 'challenges',
+          fields: [
+            { name: 'instructions', type: 'text' },
+            { name: 'timer', type: 'smallint' },
+            { name: 'autoReply', type: 'boolean' },
+            { name: 'skillIds', type: 'text []', isArray: true },
+            { name: 'skillCount', type: 'smallint', extractor: (record) => _.size(record['skillIds']) },
+            { name: 'firstSkillId', type: 'text', extractor: (record) => _.get(record['skillIds'], 0) },
+          ],
+          indices: ['firstSkillId'],
+        };
+        const fullLearningContent = {
+          challenges: [{}],
+        };
+        const learningContent = fullLearningContent[table.name];
+
+        const error = await catchErr(databaseHelper.saveLearningContent)(table, learningContent, databaseConfig);
+
```

Comment:
```suggestion
        expect(error).to.be.instanceof(PrimaryKeyNotNullConstraintError);
```

Pour moi ce n'est pas dérangeant de tester les deux ici, mais le check du type d'erreur me semble en effet le plus important.

Reply:
Fait dans https://github.com/1024pix/pix-db-replication/pull/121#discussion_r1004193048